lib/Tools/circt-lec/ConstructLEC.cpp

```diff
@@ -94,13 +119,30 @@ void ConstructLECPass::runOnOperation() {
     return signalPassFailure();
   }
 
-  // Reuse the name of the first module for the entry function, so we don't have
-  // to do any uniquing and the LEC driver also already knows this name.
+  // Only construct the miter with no additional insertions.
+  if (insertMode == lec::InsertAdditionalModeEnum::None) {
+    constructMiter(builder, loc, moduleA, moduleB);
+    return;
+  }
+
+  mlir::FailureOr<mlir::LLVM::LLVMFuncOp> printfFunc;
+  auto ptrTy = LLVM::LLVMPointerType::get(builder.getContext());
+  auto voidTy = LLVM::LLVMVoidType::get(&getContext());
+  // Lookup or declare printf function.
+  printfFunc = LLVM::lookupOrCreateFn(builder, getOperation(), "printf", ptrTy,
+                                      voidTy, true);
+  if (failed(printfFunc)) {
+    getOperation()->emitError("failed to lookup or create printf");
+    return signalPassFailure();
+  }
+
+  // Reuse the name of the first module for the entry function, so we don't
+  // have to do any uniquing and the LEC driver also already knows this name.
   FunctionType functionType = FunctionType::get(&getContext(), {}, {});
   func::FuncOp entryFunc =
       builder.create<func::FuncOp>(loc, firstModule, functionType);
 
-  if (insertMainFunc) {
+  if (insertMode == lec::InsertAdditionalModeEnum::Main) {
```

Comment:
Probably needs to wrap L164-173

Reply:
I don't think so, because the lines you mentioned should be present in 'reporting' mode

Reply:
Oops this was an accidental comment, was a reply to another one I deleted!